MoneyDataType/CurrencyConverter.cs

```diff
@@ -1,25 +1,30 @@
+using Money.Abstractions;
+using Newtonsoft.Json;
 using System;
 using System.Text.Json;
-using System.Text.Json.Serialization;
-using Money.Abstractions;
+using JsonSerializer = Newtonsoft.Json.JsonSerializer;
 
-namespace Money.Serialization
+namespace Money.Serialization;
+
+internal class CurrencyConverter : System.Text.Json.Serialization.JsonConverter<ICurrency>
 {
-    internal class CurrencyConverter : JsonConverter<ICurrency>
-    {
-        public override ICurrency Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
-            => Currency.FromISOCode(reader.GetString());
+    public override ICurrency Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options) =>
+        Currency.FromIsoCode(reader.GetString());
 
-        public override void Write(Utf8JsonWriter writer, ICurrency value, JsonSerializerOptions options)
-            => writer.WriteStringValue(value.CurrencyIsoCode);
-    }
+    public override void Write(Utf8JsonWriter writer, ICurrency value, JsonSerializerOptions options) =>
+        writer.WriteStringValue(value.CurrencyIsoCode);
+}
 
-    internal class LegacyCurrencyConverter : Newtonsoft.Json.JsonConverter<ICurrency>
-    {
-        public override ICurrency ReadJson(Newtonsoft.Json.JsonReader reader, Type objectType, ICurrency existingValue, bool hasExistingValue, Newtonsoft.Json.JsonSerializer serializer)
-            => Currency.FromISOCode(reader.ReadAsString());
+internal class LegacyCurrencyConverter : JsonConverter<ICurrency>
```

Comment:
Looking at the namespace conflicts, it would probably be best to move this class into its own file.